src/test/java/com/squareup/javapoet/TypeSpecTest.java

```diff
@@ -906,6 +906,63 @@
         + "}\n");
   }
 
+  @Test public void defaultModifiersForInterfaceMembers() throws Exception {
+    TypeSpec taco = TypeSpec.interfaceBuilder("Taco")
+        .addField(FieldSpec.builder(String.class, "SHELL")
+            .addModifiers(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL)
+            .initializer("$S", "crunchy corn")
+            .build())
+        .addMethod(MethodSpec.methodBuilder("fold")
+            .addModifiers(Modifier.PUBLIC, Modifier.ABSTRACT)
+            .build())
+        .addType(TypeSpec.classBuilder("Topping")
+            .addModifiers(Modifier.PUBLIC, Modifier.STATIC)
+            .build())
+        .build();
+    assertThat(toString(taco)).isEqualTo(""
+        + "package com.squareup.tacos;\n"
+        + "\n"
+        + "import java.lang.String;\n"
+        + "\n"
+        + "interface Taco {\n"
+        + "  String SHELL = \"crunchy corn\";\n"
+        + "\n"
+        + "  void fold();\n"
+        + "\n"
+        + "  class Topping {\n"
+        + "  }\n"
+        + "}\n");
+  }
+
+  @Test public void defaultModifiersForMemberInterfacesAndEnums() throws Exception {
+    TypeSpec taco = TypeSpec.classBuilder("Taco")
+        .addType(TypeSpec.classBuilder("Meat")
+            .addModifiers(Modifier.STATIC)
+            .build())
+        .addType(TypeSpec.interfaceBuilder("Tortilla")
+            .addModifiers(Modifier.STATIC)
+            .build())
+        .addType(TypeSpec.enumBuilder("Topping")
+            .addModifiers(Modifier.STATIC)
```

Comment:
Ah, cool. That's what I get for phone-based reviews.
On Jan 18, 2015 11:28 AM, "Jesse Wilson" <notifications@github.com> wrote:

> In src/test/java/com/squareup/javapoet/TypeSpecTest.java
> <https://github.com/square/javapoet/pull/187#discussion-diff-23136122>:
>
> > +        + "\n"
> > +        + "  class Topping {\n"
> > +        + "  }\n"
> > +        + "}\n");
> > +  }
> > +
> > +  @Test public void defaultModifiersForMemberInterfacesAndEnums() throws Exception {
> > +    TypeSpec taco = TypeSpec.classBuilder("Taco")
> > +        .addType(TypeSpec.classBuilder("Meat")
> > +            .addModifiers(Modifier.STATIC)
> > +            .build())
> > +        .addType(TypeSpec.interfaceBuilder("Tortilla")
> > +            .addModifiers(Modifier.STATIC)
> > +            .build())
> > +        .addType(TypeSpec.enumBuilder("Topping")
> > +            .addModifiers(Modifier.STATIC)
>
> Ie. we don't do containsAll() checks with asMemberModifiers.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/square/javapoet/pull/187/files#r23136122>.
>